design-proposals/sidecar-hooks.md

```diff
@@ -0,0 +1,56 @@
+# Sidecar hooks
+
+## Overview
+
+A hooking mechanism that allows customizations to the VMI before it starts without changing KubeVirt
+core components. The common user of those hooks are Sidecars that will run in the same Pod that the
+VMI will be deployed.
+
+## Motivation
+
+A way to accommodate use cases that we might not want to bring and support in KubeVirt itself or to
+give room to test and develop features while they are still not mature enough.
+
+## Goals
+
+- Provide and maintain a set of gRPC APIs for users to do modifications to the VMI
```

Comment:
I think these are too specific, possibly even touching the implementation.

I guess that the main goal is to define a stable integration point for custom programs, that are looking to extend Kubevirt VM functionality.

Reply:
Hum, yes. In the end, it is an interface to make changes. I've changed it to the following:

```diff
-- Provide and maintain a set of gRPC APIs for users to do modifications to the VMI
+- Provide and maintain an interface for custom applications to do modifications to the VMI
```